### PR TITLE
Potential Improvement with ShowAllAccInfo() function

### PR DESCRIPTION
**Pull request by elee02**
This pull request aims to improve the ShowAllAccInfo() function in the AccountHandler class. The current implementation has a bug where it doesn't handle NULL account objects correctly, resulting in potential crashes or incorrect output.

The proposed changes include adding a check for NULL account objects before calling the ShowAccInfo() function. Additionally, a variable "i" is introduced to keep track of the number of valid accounts found. If no valid accounts are found, a message is displayed indicating that there are no existing accounts.

These changes ensure that the ShowAllAccInfo() function behaves correctly and provides accurate information about all existing accounts.

Fixes #0

### CODE REVIEW

**Comment by dilshod-obidov** [comment]
```cpp
void AccountHandler::ShowAllAccInfo(void) const{
    int i = 0;
    for(int i=0; i<m_accNum; i++){
        m_accArr[i]->ShowAccInfo(); // extra line
        if (m_accArr[i] != NULL) {
            m_accArr[i]->ShowAccInfo();
            i++;
        }
    }
    if (i == 0) {
        cout << "There is no existing account." << endl;
    }
}
```

I think that there is an extra line before if statement in the for loop: 'm_accArr[i]->ShowAccInfo();'

**Comment by elee02** [comment]
Never saw where that came from, haha ,)

File: WeeklySummary/Week12/ex3/AccountHandler.cpp [hunk 1]
```diff
@@ -102,8 +102,15 @@ void AccountHandler::WithdrawMoney(void){
 }
 
 void AccountHandler::ShowAllAccInfo(void) const{
+    int i = 0;
     for(int i=0; i<m_accNum; i++){
-        m_accArr[i]->ShowAccInfo();
```
Comment (dilshod-obidov) [inline WeeklySummary/Week12/ex3/AccountHandler.cpp:106]: extra line

**Review by dilshod-obidov — COMMENTED** [review]
reviewed

**Pull request closed without merging**